### PR TITLE
Issue 15

### LINKED ISSUE #15: Analysis of the structure for the identification of sources
Currently it library uses by default way with setup any of function by developer hand. Need analys structures before setuping for find functions which need to preparing this structures. Like this:
```golang
MyStruct struct {
   Field1 int32 `cf_hocon:" node = test, path = field "cf_default:" 0 "`
}
```
that need find 2 tag name: cf_hocon and cf_default and after using only 2 functions for configuring structure

### PR DESCRIPTION
**Pull request by kubitre**
resolve #15

### CODE REVIEW

File: lib.go [hunk 1]
```diff
@@ -10,11 +11,11 @@ ConfigureEasy - default pipeline setup for configure your structure
...
-	return pipeline.Configure(structure, fileName, []pipeline.FuncType{
-		pipeline.FunctionSetupEnvironment,
-		pipeline.FunctionSetupHocon,
-		pipeline.FunctionSetupDefault,
-	}, pipeline.EmptyAdditionalPrefix)
+	return pipeline.Configure(structure, fileName, []infra.FuncType{
+		infra.FunctionSetupEnvironment,
+		infra.FunctionSetupHocon,
+		infra.FunctionSetupDefault,
+	}, pipeline.EmptyAdditionalPrefix, pipeline.DurtyConfiguring)
```
Comment (artemkaxboy) [inline lib.go]: It's better to use RawConfiguration, or at least spell it dirty

File: tags/analys_func.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+	"github.com/goreflect/gostructor/infra"
+)
+
+/*
+GetFunctionTypes - return slice of functions which should configuring sourceStruct structure
+*/
+func GetFunctionTypes(sourceStruct interface{}) []infra.FuncType {
+	summirize := []int{}
+	value := reflect.Indirect(reflect.ValueOf(sourceStruct))
```
Comment (artemkaxboy) [inline tags/analys_func.go]: It's exported function, so you should check input data, because you may get here not a pointer but direct value

Comment (kubitre) [reply tags/analys_func.go]: i have get direct value, i don't have pointer =)

Comment (kubitre) [reply tags/analys_func.go]: I want to get the value I don't need a pointer here. If you read the documentation, whatever I transmit, whether it is a value or a link, in the output I will get a value of type reflect.Value
code from value.go:
```golang
// Indirect returns the value that v points to.
// If v is a nil pointer, Indirect returns a zero Value.
// If v is not a pointer, Indirect returns v.
func Indirect(v Value) Value {
	if v.Kind() != Ptr {
		return v
	}
	return v.Elem()
}
```

File: lib.go (hunk 1 above)
Comment (kubitre) [reply lib.go]: ++

**Pull request merged**